### ISSUE TITLE
Sorting based on edits / page views / likes shows only 1 row on the first page of wikis

### ISSUE THREAD

**Issue by Manasa2850** [publiclab/plots2#9290]
### Please describe the problem (or idea)
When we sort wikis based on `title` or `last edited`, it works fine and shows the same number of rows on the first page before and after sorting.
However, if we sort based on `edits`, `pages views` or `likes`, it shows only one row on the first page.

### Please show us where to look
https://publiclab.org/wiki

![ezgif com-video-to-gif](https://user-images.githubusercontent.com/43354059/110972466-09e63100-8382-11eb-983c-51c1eeb9845e.gif)

**Comment by gaurav2699** [comment]
Hey @Manasa2850, I have just started out contributing to open source and already contributed to a first-timers issue. I want to try out this issue.

**Comment by Manasa2850** [comment]
Hi @gaurav2699 sure you can work on it.
Even I'm not able to see the bug locally. I observed this on https://publiclab.org/wiki. Maybe @cesswairimu or @jywarren can give us more inputs on this.

**Comment by jywarren** [comment]
Hi, this is a good catch, but I wonder how we can find this bug in more detail. I wonder - should we try writing a functional test for it, by building on these tests?

https://github.com/publiclab/plots2/blob/c9349cae8af3e8ed2e27c77d9e35173a6cfa9194/test/functional/wiki_controller_test.rb#L20-L32

Could we try to open a PR that "catches" the bug by counting the # of results on each page? 

Thank you for your help!!!

**Comment by Manasa2850** [comment]
@jywarren I'm not able to make a PR to catch that bug since I don't see it locally. I observed it on https://publiclab.org/wiki. It would be great if someone who's observing this locally too, could make the PR.
Thanks!

**Comment by gaurav2699** [comment]
@jywarren any reason why this bug is not replicated locally?

**Comment by cesswairimu** [comment]
This is really weird..I think it has smth to do with pagination -- the no. of elements per page not being clearly defined, on the website I have noticed when you sort with title the number per page reduces to 8, if you go to page 2 its 3 etc the number per page keeps shifting...locally  I cannot replicate too but also it doesn't paginate for me...I created upto 20 wikis and they are all showing on the first page

**Comment by gaurav2699** [comment]
Hey, @cesswairimu @jywarren I made a PR for this which should fix this issue. Basically, I specified the number of items on every page to be 10. If I could replicate this bug in my local development then I would be really sure if this bug is fixed but hopefully, it does. Please review

**Comment by cesswairimu** [comment]
The changes will appear on https://stable.publiclab.org/  in 15 min or so...lets confirm there this is all fixed. Thanks everyone :heart:

**Comment by cesswairimu** [comment]
Looks like the issue is not fixed yet :disappointed: https://stable.publiclab.org/wiki?sort=page_views reopening this

**Comment by gaurav2699** [comment]
This is really weird, as it was working with no issues in my local development. Maybe pagination was not the problem, or maybe we are missing something.

**Comment by cesswairimu** [comment]
yeah pagination now works fine on my local too...I agree we may be missing smth

**Comment by Manasa2850** [comment]
@cesswairimu another thing I observed is that the same entry appears in multiple rows. For example, `sandbox-soil` appears in both row 1 and row 2. But ideally it should appear in only one of the rows. This is why the number of rows is increasing to 1031. Not sure why this is happening.

**Comment by cesswairimu** [comment]
oh wow, great observation...maybe we need to add "distinct" on the query fetching on index :thinking: ...I will investigate more on this and revert

**Comment by ebarry** [comment]
Fascinating!!! I am seeing this also. 

I captured a gif that shows how even clicking through multiple pages, only the /barnstars page is shown. I happen to like this page, but, it's really incorrect as a result. 

![sorting-by-likes](https://user-images.githubusercontent.com/161439/128883882-92962106-f0a5-4f9d-a17e-5f15e202ab2e.gif)
